### DISCUSSION TITLE
Found a Config but seems the format has changed, help please.

### DISCUSSION THREAD

**Discussion by KAGEYAM4** [sezanzeb/input-remapper#724] [Q&A] [answered]
This is the [Capslock+](https://github.com/wo52616111/capslock-plus/issues/90) whose functionality i am trying to replicate ( if you know already existing project for GNU/linux. Please let me know )

This the config a user gave on that above link, but when i compared the configs to `input-remapper2` - it is not matching.

<details>
<summary>Config</summary>

```
{
    "mapping": {
        "1,58,1+1,32,1": [
            "Down",
            "keyboard"
        ],
        "1,58,1+1,31,1": [
            "Left",
            "keyboard"
        ],
        "1,58,1+1,33,1": [
            "Right",
            "keyboard"
        ],
        "1,58,1+1,17,1": [
            "BackSpace",
            "keyboard"
        ],
        "1,58,1+1,25,1": [
            "Home",
            "keyboard"
        ],
        "1,58,1+1,39,1": [
            "End",
            "keyboard"
        ],
        "1,58,1+1,36,1": [
            "Shift_L+Left",
            "keyboard"
        ],
        "1,58,1+1,23,1": [
            "Shift_L+Up",
            "keyboard"
        ],
        "1,58,1+1,37,1": [
            "Shift_L+Down",
            "keyboard"
        ],
        "1,58,1+1,38,1": [
            "Shift_L+Right",
            "keyboard"
        ],
        "1,58,1+1,51,1": [
            "Alt_L+Shift_L+Right",
            "keyboard"
        ],
        "1,58,1+1,35,1": [
            "Control_L+Shift_L+Left",
            "keyboard"
        ],
        "1,58,1+1,30,1": [
            "Control_L+Left",
            "keyboard"
        ],
        "1,58,1+1,34,1": [
            "Control_L+Right",
            "keyboard"
        ],
        "1,58,1+1,18,1": [
            "Up",
            "keyboard"
        ],
        "1,58,1+1,14,1": [
            "k(Home).m(Shift_L, k(End)).k(BackSpace)",
            "keyboard"
        ],
        "1,58,1": [
            "if_single(key(Control_L), , timeout=300)",
            "keyboard"
        ],
        "1,58,1+1,22,1": [
            "Shift_L+Home",
            "keyboard"
        ],
        "1,58,1+1,24,1": [
            "Shift_L+End",
            "keyboard"
        ],
        "1,58,1+1,52,1": [
            "Control_L+Shift_L+Right",
            "keyboard"
        ],
        "1,58,1+1,57,1": [
            "KEY_ENTER",
            "keyboard"
        ],
        "1,58,1+1,12,1": [
            "Control_L+KEY_PAGEUP",
            "keyboard"
        ],
        "1,58,1+1,13,1": [
            "Control_L+KEY_PAGEDOWN",
            "keyboard"
        ],
        "1,58,1+1,46,1": [
            "Control_L+c",
            "keyboard"
        ],
        "1,58,1+1,47,1": [
            "Control_L+v",
            "keyboard"
        ],
        "1,58,1+1,45,1": [
            "Control_L+x",
            "keyboard"
        ],
        "1,58,1+1,44,1": [
            "Control_L+z",
            "keyboard"
        ]
    }
}
```
</details>

Also Extra Question --> while creating `input -> output` mapping `screenshot below` , i have keys like `capslock` `alt` and there combination with other alphabet keys i am trying to map. But `capslock` and `alt` gets send on there own to the window and moves the focus away. How to prevent it ?

<details>
<summary>Screenshot</summary>

![image](https://github.com/sezanzeb/input-remapper/assets/75798544/f55d2755-be63-4243-90b2-73c101bdbd95)

</details>

**Comment by jonasBoss** [comment, marked as answer]
yeah with Input-remapper 2.0 the save file format has hanged completely.
Input-remapper can migrate old save files from version 1.x to 2.x
You can do this by renaming (backup) the `~/.config/input-remapper-2` folder and then place the mapping in a folder `~/.config/input-remapper/presets/Device Name/my_preset.json`. Also create a new file:
`~/.config/input-remapper/config.json`
```json
{
    "version": "1.5.0"
}
```

When you now start input-remapper it should migrate the mapping to the new format. This will create a new `~/.config/input-remapper-2` folder. You can now move the backup into that folder.

**Reply by KAGEYAM4** [reply]
Thank you it worked, also one extra thing can you please help with this ->

while creating input -> output mapping screenshot below , i have keys like capslock alt and there combination with other alphabet keys i am trying to map. But capslock and alt gets send on there own to the window and moves the focus away. How to prevent it ?

**Reply by jonasBoss** [reply]
you can try the [`if_single`](https://github.com/sezanzeb/input-remapper/blob/main/readme/macros.md#if_single) macro. If you make a mapping `ALT_L`->`if_single(key(ALT_L), hold(ALT_L))` this basically delays the `ALT_L` key until you either let go of that key or another key is pressed. It might cause issues with your other mappings, that is something you need to try, and please let us know if it works.
